dbms/src/Storages/MergeTree/MergeTreeDataPart.cpp

```diff
@@ -750,6 +749,8 @@ void MergeTreeDataPart::loadColumns(bool require)
         return;
     }
 
+    is_frozen = !poco_file_path.canWrite();
```

Comment:
Note: will slow down startup of clickhouse-server, because `stat` system call is performed twice.
Ok.